main.py

```diff
@@ -48,177 +62,352 @@ def validate_yaml_file(yaml_path: Path) -> dict:
+            )
 
         # Validate distance
         approved_distances = {0, 5, 10, 25, 50, 100}
-        if parameters['distance'] not in approved_distances:
-            raise ConfigError(f"Invalid distance value in config file {config_yaml_path}. Must be one of: {approved_distances}")
+        if parameters["distance"] not in approved_distances:
+            logger.error(
+                f"Invalid distance value '{parameters['distance']}'. Must be one of {approved_distances}."
+            )
+            raise ConfigError(
+                f"Invalid distance value in config file {config_yaml_path}. Must be one of: {approved_distances}"
+            )
 
         # Ensure blacklists are lists
-        for blacklist in ['company_blacklist', 'title_blacklist','location_blacklist']:
+        for blacklist in ["company_blacklist", "title_blacklist", "location_blacklist"]:
             if not isinstance(parameters.get(blacklist), list):
-                raise ConfigError(f"'{blacklist}' must be a list in config file {config_yaml_path}")
+                raise ConfigError(
+                    f"'{blacklist}' must be a list in config file {config_yaml_path}"
+                )
             if parameters[blacklist] is None:
                 parameters[blacklist] = []
 
+        logger.info("Configuration validated successfully.")
         return parameters
 
     @staticmethod
-    def validate_secrets(secrets_yaml_path: Path) -> str:
+    def validate_secrets(secrets_yaml_path: Path) -> tuple:
+        """
+        Validates the secrets file to ensure required keys are present and valid.
+        """
         secrets = ConfigValidator.validate_yaml_file(secrets_yaml_path)
-        mandatory_secrets = ['llm_api_key']
+        mandatory_secrets = ["email", "password", "llm_api_key"]
 
         for secret in mandatory_secrets:
             if secret not in secrets:
-                raise ConfigError(f"Missing secret '{secret}' in file {secrets_yaml_path}")
+                raise ConfigError(
+                    f"Missing secret '{secret}' in file {secrets_yaml_path}"
+                )
+
+        if not ConfigValidator.validate_email(secrets["email"]):
+            raise ConfigError(
+                f"Invalid email format in secrets file {secrets_yaml_path}."
+            )
+        if not secrets["password"]:
+            raise ConfigError(
+                f"Password cannot be empty in secrets file {secrets_yaml_path}."
+            )
+        if not secrets["llm_api_key"]:
+            raise ConfigError(
+                f"llm_api_key cannot be empty in secrets file {secrets_yaml_path}."
+            )
+        return secrets["email"], str(secrets["password"]), secrets["llm_api_key"]
 
-        if not secrets['llm_api_key']:
-            raise ConfigError(f"llm_api_key cannot be empty in secrets file {secrets_yaml_path}.")
-        return secrets['llm_api_key']
 
 class FileManager:
```

Comment:
can to take these classes out. as you made lot of refactoring delta, why don't we do it complete.

Reply:
done